packages/fast-components/src/dialog/dialog.ts

```diff
@@ -0,0 +1,145 @@
+import { attr, FastElement } from "@microsoft/fast-element";
+import { keyCodeEscape, keyCodeTab } from "@microsoft/fast-web-utilities";
+import tabbable from "tabbable";
+
+export class Dialog extends FastElement {
+    @attr
+    public modal: boolean = true;
+
+    @attr
+    public hidden: boolean = false;
+
+    @attr({ attribute: "trap-focus" })
+    public trapFocus: boolean = true;
+
+    @attr({ attribute: "aria-describedby" })
+    public ariaDescribedby: string;
+
+    @attr({ attribute: "aria-labelledby" })
+    public ariaLabelledby: string;
+
+    @attr({ attribute: "aria-label" })
+    public ariaLabel: string;
+
+    public dialog: HTMLDivElement;
+
+    public dismiss(): void {
+        this.dispatchEvent(
+            new CustomEvent("dismiss", {
+                bubbles: true,
+                composed: true,
+            })
+        );
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        document.addEventListener("keydown", this.handleDocumentKeydown);
+
+        if (this.shouldDialogTrapFocus()) {
+            document.addEventListener("focusin", this.handleDocumentFocus);
+
+            if (this.shouldForceFocus(document.activeElement)) {
+                this.focusFirstElement();
+            }
+        }
+    }
+
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+
+        document.removeEventListener("keydown", this.handleDocumentKeydown);
+
+        if (this.shouldDialogTrapFocus()) {
+            document.removeEventListener("focusin", this.handleDocumentFocus);
+        }
+    }
+
+    private handleDocumentKeydown = (e: KeyboardEvent): void => {
+        // remove as any here when we have boolean support
```

Comment:
Is this comment accurate? Seems like it might have been misplaced.

Reply:
It use to be accurate. :) I thought I got most of them.